src/index.rs

```diff
@@ -293,3 +407,141 @@ fn index_single_block(
+                    script_sig: i.script_sig.to_bytes(),
+                    txinwitness: i.witness.to_vec(),
+                    script_pub_key: prevout.script_pubkey.to_bytes(),
+                }) {
+                    Ok(Some(pubkey_from_input)) => match pubkey_from_input {
+                        crate::sp::PubKeyFromInput::XOnlyPublicKey(xonly_pubkey) => xonly_pubkeys.push(xonly_pubkey),
+                        crate::sp::PubKeyFromInput::PublicKey(pubkey) => pubkeys.push(pubkey), 
+                    },
+                    Ok(None) => (),
+                    Err(_) => panic!("Scanning for public keys failed for tx: {}", txid),
+                }
+            }
+            let pubkeys_ref: Vec<&PublicKey> = pubkeys.iter().collect();
+            let pubkeys_ref = pubkeys_ref.as_slice();
+
+            // if the pubkeys have opposite parity, the combine_pubkey_result will be Err
+            let combine_pubkey_result = PublicKey::combine_keys(pubkeys_ref);
+
+            if combine_pubkey_result.is_ok() && (!pubkeys.is_empty() || !xonly_pubkeys.is_empty()) {
+
+                let pubkeys_ref: Vec<&PublicKey> = pubkeys.iter().collect();
+                let input_pub_keys = if pubkeys.is_empty() {
+                    None
+                } else {
+                    Some(pubkeys_ref.as_slice())
+                }; 
+
+                let xonly_pubkeys_ref: Vec<&XOnlyPublicKey> = xonly_pubkeys.iter().collect();
+                let input_xpub_keys = if xonly_pubkeys.is_empty() {
+                    None
+                } else {
+                    Some(xonly_pubkeys_ref.as_slice())
+                }; 
+                
+                let tweak = crate::sp::recipient_calculate_tweak_data(input_xpub_keys, input_pub_keys, &outpoints).unwrap();
+
+                if let Some(block_hash) = self.index.filter_by_txid(txid).next() {
+                    if let Some(value) = self.map.get_mut(&block_hash) {
+                        value.extend(tweak.iter());
+                    } else {
+                        self.map.insert(block_hash, Vec::from_iter(tweak)); 
+                    }
+                } else {
+                    panic!("Unexpected unknown transaction");
+                }
+            }
+
+            ControlFlow::Continue(())
+        }
+    }
+
+    let mut map: HashMap<BlockHash, Vec<u8>> = HashMap::with_capacity(index.batch_size);
+    let mut index_block = IndexBlockVisitor {
+        daemon,
+        index,
+        map: &mut map
+    };
+    bsl::Block::visit(&block, &mut index_block).expect("core returned invalid block");
+    for (hash, tweaks) in map {
+        let height = index.chain.get_block_height(&hash).expect("Unexpected non existing blockhash");
+        let mut value: Vec<u8> = u64::try_from(height).expect("Unexpected invalid usize").to_be_bytes().to_vec();
+        value.extend(tweaks.iter());
+        batch.tweak_rows.push(value);
```

Comment:
If I understand correctly, the key of the stored key-value pair is the block height. Is RocksDB the right tool for the job then? I think this can easily be achieved using two files that only need appending when building/updating the index: one for the tweaks, and one for the offsets and lengths in the tweaks file, for every block (for random access).